app/src/main/kotlin/io/intrepid/skotlinton/base/BasePresenter.kt

```diff
@@ -8,56 +12,38 @@ import io.reactivex.Observable
 import io.reactivex.Scheduler
 import io.reactivex.Single
 import io.reactivex.disposables.CompositeDisposable
+import timber.log.Timber
 
-abstract class BasePresenter<V : BaseContract.View>(protected var view: V?, configuration: PresenterConfiguration) : BaseContract.Presenter {
+abstract class BasePresenter<S : BaseScreen>(protected var screen: S?, configuration: PresenterConfiguration) :  ViewModel(), LifecycleObserver {
 
     protected val ioScheduler: Scheduler = configuration.ioScheduler
     protected val uiScheduler: Scheduler = configuration.uiScheduler
     protected val userSettings: UserSettings = configuration.userSettings
     protected val restApi: RestApi = configuration.restApi
     protected val crashReporter: CrashReporter = configuration.crashReporter
-
     protected val disposables: CompositeDisposable = CompositeDisposable()
 
-    private var isViewBound = false
-
-    override fun onViewCreated() {
-
+    init {
+        Timber.e("Initialize BasePresenter")
     }
 
-    override fun bindView(view: BaseContract.View) {
-        @Suppress("UNCHECKED_CAST")
-        this.view = view as V
-        if (!isViewBound) {
-            onViewBound()
-            isViewBound = true
-        }
+    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
+    fun onCreateScreenLifecycle() {
+        Timber.v("Presenter onCreateScreenLifecycle: $this")
+        screenCreated()
```

Comment:
I'm not sure why, but this breaks functionality in the Example2Presenter. Although Example2Presenter overrides `screenCreated()`, I'm not seeing it be called executed and the Example2 screen is not showing the user's IP address as a result.

Reply:
I haven't checked it on this branch but there's a UI test that should hit this